starfish/imagestack/imagestack.py

```diff
@@ -711,18 +714,53 @@ def transform(
             group_by = {Indices.X, Indices.Y}
 
         indices = list(self._iter_indices(group_by))
+        slice_list = [self._build_slice_list(index)[0]
+                      for index in indices]
 
+        indices_and_slice_list = zip(indices, slice_list)
         if verbose:
-            tiles = tqdm(self._iter_tiles(indices))
-        else:
-            tiles = self._iter_tiles(indices)
+            indices_and_slice_list = tqdm(indices_and_slice_list)
 
-        applyfunc: Callable = partial(func, **kwargs)
+        applyfunc: Callable = partial(
+            self._multiprocessing_workflow,
+            _StripArguments(partial(func, **kwargs), (1, 2)),
```

Comment:
"remove 2nd and 3rd arguments of this partial, which probably _come from kwargs_" is a pretty indecipherable piece of code. :|

Reply:
haha, it's actually more complicated than that.  arguments 2 & 3 come from `_multiprocessing_workflow`.  kwargs are bound, so they can't be stripped.